common/src/Model/Brush.cpp

```diff
@@ -947,8 +970,23 @@ namespace TrenchBroom {
             assert(remaining.vertexCount() + moving.vertexCount() == vertexCount());
             
             // Will vertices be removed?
-            if (!allowVertexRemoval && result.vertexCount() < m_geometry->vertexCount())
-                return false;
+            if (!allowVertexRemoval) {
+                if (result.vertexCount() < m_geometry->vertexCount())
+                    return false;
+                
+                // All moving vertices must still be present in the result
+                for (const Vec3& movingVertex : moving.vertexPositions()) {
```

Comment:
If the previous check holds, this should also hold, so it's superfluous, or am I missing something?